packages/@azure/servicebus/data-plane/test/stress/messageRandomDispositionOnSession.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  ServiceBusClient,
+  SendableMessageInfo,
+  OnMessage,
+  OnError,
+  delay,
+  ServiceBusMessage
+} from "../../lib";
+
+const connectionString = "";
+const queueName = "";
+
+const testDurationInMilliseconds = 60000 * 5 * 12 * 24 * 7; // 1 week
+
+const messagesToProcess: Set<number> = new Set<number>();
+const messageAbandonedMap: { [key: number]: number } = {};
+
+let abandonCount = 0;
+let completeCount = 0;
+let deadletterCount = 0;
+let deferCount = 0;
+
+let msgId = 1;
+
+let snapshotIntervalID: any;
+
+let isJobDone = false;
+
+async function main(): Promise<void> {
+  snapshotIntervalID = setInterval(snapshot, 5000); // Every 5 seconds
+  sendMessages();
+  receiveMessages();
+}
+
+async function sendMessages(): Promise<void> {
+  const ns = ServiceBusClient.createFromConnectionString(connectionString);
+  const client = ns.createQueueClient(queueName);
+  try {
+    const sender = client.createSender();
+
+    while (!isJobDone) {
+      const message: SendableMessageInfo = {
+        messageId: msgId,
+        body: "test",
+        label: `${msgId}`,
+        sessionId: "session-1"
+      };
+      messageAbandonedMap[msgId] = 0;
+      messagesToProcess.add(msgId);
+      msgId++;
+      await sender.send(message);
+      await delay(2000); // Throttling send to not increase queue size
+    }
+  } finally {
+    client.close();
+    ns.close();
+  }
+}
+
+async function receiveMessages(): Promise<void> {
+  const ns = ServiceBusClient.createFromConnectionString(connectionString);
+  const client = ns.createQueueClient(queueName);
+
+  try {
+    const receiver = await client.createSessionReceiver({ sessionId: "session-1" });
+    const onMessageHandler: OnMessage = async (brokeredMessage: ServiceBusMessage) => {
+      const receivedMsgId = brokeredMessage.messageId;
+
+      if (typeof receivedMsgId !== "number") {
+        throw new Error("MessageId is corrupt or is of unexpected type");
+      }
+
+      const seed = Math.floor((Math.random() * 10) % 4);
+
+      switch (seed) {
+        case 0: {
+          const currCount = messageAbandonedMap[receivedMsgId];
+          if (currCount === 10) {
+            abandonCount++;
+            if (messagesToProcess.has(receivedMsgId)) {
+              messagesToProcess.delete(receivedMsgId);
+            }
+          }
+          messageAbandonedMap[receivedMsgId] = currCount + 1;
+          brokeredMessage.abandon();
```

Comment:
Yes, that is the intent. The expectation is that depending on seed value it would randomly enter this block and abandon the message.
Could you please elaborate on concern, or what were you expecting to see here?

Reply:
The same question as above -- since the message gets abandoned the first time it goes through this code, do we need the code that checks the count?

Reply:
One of the verification criteria we had captured in requirements was to verify message count and content and be able to troubleshoot issues with map build up if it happens. 
Tracking the delivery count was to help keep track of the correct number of messages in flight to help with the verification aspects of the test.
Though the chances of seed hitting abandon code path 10 times is tiny, it's not non-zero.